applications/GeoMechanicsApplication/custom_utilities/process_utilities.cpp

```diff
@@ -45,4 +73,30 @@ std::vector<std::reference_wrapper<ModelPart>> ProcessUtilities::GetModelPartsFr
     return result;
 }
 
+void ProcessUtilities::AddProcessesSubModelPartList(const Parameters& rProjectParameters, Parameters& rSolverSettings)
+{
+    std::unordered_set<std::string, StringHash, std::equal_to<>> domain_condition_names;
+    const auto root_name = rSolverSettings["model_part_name"].GetString();
+    const auto prefix    = root_name + ".";
+
+    const std::vector<std::string> process_lists_to_be_checked = {
+        "constraints_process_list", "loads_process_list", "auxiliary_process_list"};
+    if (rProjectParameters.Has("processes")) {
+        const auto processes = rProjectParameters["processes"];
+        for (const auto& r_process_list_name : process_lists_to_be_checked) {
+            if (processes.Has(r_process_list_name))
+                ExtractModelPartNames(processes[r_process_list_name], domain_condition_names, root_name, prefix);
+        }
```

Comment:
To elaborate on my previous suggestion, we could probably do something like this here (although I'm not sure if the syntax is completely correct):
```suggestion
        for (const auto& r_process_list : rProjectParameters["processes"]) {
                ExtractModelPartNames(r_process_list , domain_condition_names, root_name, prefix);
        }
```

Reply:
It works ;)